pkg/ingester/errors.go

```diff
@@ -22,10 +22,17 @@ import (
 	"github.com/grafana/mimir/pkg/util/validation"
 )
 
+const (
+	unavailable = int(codes.Unavailable)
+)
+
 var (
 	// This is the closest fitting Prometheus API error code for requests rejected due to limiting.
 	tooBusyError = httpgrpc.Errorf(http.StatusServiceUnavailable,
```

Comment:
If you have a list of things to fix, I'd love for ingesters to _stop_ returning HTTP status codes for their gRPC calls and instead use gRPC codes.

Reply:
In internal Slack discussion, I was proposing to return semantic errors in `WriteResponse` instead. I don't think gRPC codes map nicely onto errors in ingesters. I agree that our current approach is terrible.

Reply:
This is a more complicated story right now. We will definitely need a better solution for this. I agree that the current approach is not the best, but in this moment it is not safe to change this.